### PR TITLE
(examples) Usage of retryExchange

### PR DESCRIPTION
**Pull request by yankovalera**
Provides a React example describing the way to initialise the urlq client with retryExchange to retry queries if an error occurred

### CODE REVIEW

File: examples/react-urql-retry/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Integrating with React
+
+Integrating urql is as simple as:
+
+1.  Install packages
+
+yarn add urql @urql/exchange-retry graphql
```
Comment (JoviDeCroock) [inline examples/react-urql-retry/README.md:7]: Let's wrap this one in backticks as well, makes it easier for folks to copy-paste

File: examples/react-urql-retry/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+# or
+
+npm install --save urql @urql/exchange-retry graphql
+
+2.  Setting up the Client [here](src/App.js)
```
Comment (JoviDeCroock) [inline examples/react-urql-retry/README.md:13]:
```suggestion
2.  Setting up the Client and adding the `retryExchange` [here](src/App.js)
```

**Review by JoviDeCroock — APPROVED** [review]
few nits, looks great!

File: .gitignore [hunk 3]
```diff
@@ -16,3 +16,9 @@ exchanges/*/LICENSE
 # TODO: Figure out how to remove these:
 tmp/
 dist/
+examples/react-urql-retry/public/favicon.ico
```
Comment (kitten) [inline .gitignore:19]: Same suggestions as per: https://github.com/FormidableLabs/urql/pull/1573#pullrequestreview-646280627
(i.e. `examples/with-retry` naming and unified `gitignore` changes)

File: examples/react-urql-retry/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Integrating with React
```
Comment (kitten) [inline examples/react-urql-retry/README.md:1]:
```suggestion
# Integrating `@urql/exchange-retry`’s retryExchange
```

File: examples/react-urql-retry/package.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "urlq-retry-query",
+  "version": "1.0.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.12.0",
+    "@testing-library/react": "^11.2.6",
+    "@testing-library/user-event": "^12.8.3",
```
Comment (kitten) [inline examples/react-urql-retry/package.json:8]: nit: Extraneous dependencies; also as the other one, `web-vitals` is also extraneous and `react-scripts` to be a minor range and to go to `devDependencies`

File: examples/react-urql-retry/src/App.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import "./App.css";
+import Color from "./pages/Color";
+
+const client = createClient({
+  url: "https://trygql.dev/graphql/intermittent-colors",
+  exchanges: [
+    retryExchange({
+      maxNumberAttempts: 5,
+      retryIf: error => {
+        return Boolean(error && (error.graphQLErrors.length > 0 || error.networkError));
```
Comment (kitten) [inline examples/react-urql-retry/src/App.js:13]: Just copying this over from your other PR 🙌 

```suggestion
        return error.graphQLErrors.length || error.networkError;
```

which actually can be simplified to `retryIf: () => true` to be fair, but it's still worth having this here for the example. If you wanted to be very precise, you could add:

```suggestion
        return error.graphQLErrors.some(x => x.extensions?.code === 'NO_SOUP') || !!error.networkError;
```

File: examples/react-urql-retry/src/App.test.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
```
Comment (kitten) [inline examples/react-urql-retry/src/App.test.js:4]: Just as the above says, extraneous files as we won't need tests added for the example. I assume they're from the boilerplate

File: examples/react-urql-retry/src/pages/Color.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    <div>
+      {fetching && <p>Loading...</p>}
+
+      {error && <p>Oh no... {error.message}</p>}
+
+      {data && (
+        <div style={{ backgroundColor: data.randomColor.hex }}>
+          {data.randomColor.name}
+        </div>
+      )}
```
Comment (kitten) [inline examples/react-urql-retry/src/pages/Color.js:28]: One nice thing you could add to differentiate it from the other example is something like:

```suggestion
      )}
      
      {result.operation && (
        <p>We retried {res.operation.context.retryCount} times to get a result without an error.</p>
      )}
```

File: examples/react-urql-retry/src/App.test.js (hunk 7 above)
Comment (yankovalera) [reply examples/react-urql-retry/src/App.test.js:4]: yes, removed!

**Pull request merged**